docs/pallas/grid_blockspec.md

```diff
@@ -233,3 +234,44 @@ See [the Pallas TPU documentation](https://jax.readthedocs.io/en/latest/pallas/t
  [309 309 309 319 319 319]]
 
 ```
+
+The behavior documented above applies to the `indexing_mode=pl.Blocked()`.
```

Comment:
Should we have section headers to make it clearer that we transition to a different mode?

Reply:
I added a section header for this. We should revisit the structure if and when we are adding more material here.